### PR TITLE
RCTL/RACCT Support

### LINKED ISSUE #8: RCTL/RACCT statistics
Import the `rctl(8)` code from [phyber/jail_exporter](https://github.com/phyber/jail_exporter). I guess the API could be something along the lines of:

```rust
fn RunningJail::resource_usage(&self) -> Result<HashMap<String,i64>,JailError>
```

If `RCTL`/`RACCT` are unavailable or disabled, an appropriate `JailError` should be returned.

### PR DESCRIPTION
**Pull request by fabianfreyer**
Closes #8 

- Implement wrappers for RCTL structures (moved to [fubarnetes/rctl](https://github.com/fubarnetes/rctl)):
  - [X] Subject
    - [X] `impl Display`
    - [X] Parser
    - [X] Docs
  - [X] Action
    - [X] `impl Display`
    - [X] Parser
    - [X] Docs
  - [X] Resource
    - [X] `impl Display`
    - [X] Parser
    - [X] Docs
  - [X] Limit
    - [x] `impl Display`
    - [X] Parser
    - [x] Docs
  - [x] Rule
    - [x] `impl Display`
    - [x] Parser
    - [x] Docs
  - [x] Filter
    - [x] `impl Display`
    - [x] Docs
- wrap functions (moved to [fubarnetes/rctl](https://github.com/fubarnetes/rctl)):
  - [x] Check kernel `RCTL`/`RACCT` support
  - [X] Iteration over all rules matching a `Filter`
  - [X] Iteration over all rules applying to a `Subject`
  - [X] add a rule
  - [X] remove all rules matching a `Filter`
  - [x] get accounting information for a subject
- Jail Boilerplate:
  - [X] Convenient wrapper to set rules on a `StoppedJail`, so that they get applied on jail start.
  - [X] Convenient wrapper to get resource usage information

### CODE REVIEW

**Comment by fabianfreyer** [comment]
Squashed and rebased previous commits.

**Comment by fabianfreyer** [comment]
@phyber: PTAL. This should now implement most of the functionality in the `rctl` module in [jail_exporter](https://github.com/phyber/jail_exporter). Sorry about the volume of it all though :)

**Comment by phyber** [comment]
Slowly reading through this, quite a lot and my Rust isn't quite up to this level yet (as you can tell), but I think I get what's going on and it's looking good so far. :)

**Comment by fabianfreyer** [comment]
No worries, I'm still pretty new to rust too, and just winging it :). Thanks for taking a look!

Please do let me know how you'd prefer me to credit your work on [jail_exporter](https://github.com/phyber/jail_exporter) that I based this on - I guess I could set you as author on 58bb02c1deb703c251dbf4e3942c25ab99d9c4ee and ad588c34a27e4ae11289551f3d6c5109bf0ec3cf, or add a comment in the file (or feel free to PR however you would see fit)?

File: src/rctl.rs [hunk 1]
```diff
@@ -0,0 +1,1595 @@
...
+
+    // Get the filter as a C string.
+    let cfilter = CString::new(filter).map_err(Error::CStringError)?;
+
+    // C compatible output buffer.
+    let outbuflen: usize = RCTL_DEFAULT_BUFSIZE / 4;
+    let mut outbuf: Vec<i8> = vec![0; outbuflen];
+
+    // Unsafe C call to get the jail resource usage.
+    if unsafe { rctl_get_racct(cfilter.as_ptr(), filterlen, outbuf.as_mut_ptr(), outbuflen) } != 0 {
```
Comment (fabianfreyer) [inline src/rctl.rs]: from [`ractl_get_racct`(2)]:
> **RULES AND FILTERS**
> This section explains how the rule and filter format described in rctl(8) differs from the format passed to the system calls themselves.  The rctl tool provides several conveniences that the system calls do not.  When  using the system call:
>
>   *   The subject must	be fully specified.  For example, abbreviating  `user` to `u` is	not acceptable.
>   *   User and	group IDs must be numeric.  For	example, `root`	must be expressed as `0`.
>   *  Units are not permitted on resource amounts.  For example, a quantity	of 1024 bytes must be expressed	as `1024` and not `1k`.

The `fmt::Display` implementation here prints these things in the humanized form - this works well for jails, but won't work for users.

[`ractl_get_racct`(2)]: https://www.freebsd.org/cgi/man.cgi?query=rctl_get_racct&sektion=2&manpath=FreeBSD+11.2-RELEASE+and+Ports

**Comment by phyber** [comment]
A contributor note somewhere is probably good enough if you'd like to credit me. I was wondering the same for your contribution to `jail_exporter`. Since you did a PR there and you do have code in it, I should probably add you to the `Cargo.toml` authors :)

I was also pondering switching `jail_exporter` to BSD-3-Clause after reading the differences between it and MIT.

**Comment by fabianfreyer** [comment]
@phyber: don't worry about adding me to the authors - I don't think it was a very substantial contribution :) 

Is the contributor / license note on 0f1faceb172faa1639bb2df6a0c3d509204d2518 enough? If that was provided under BSD-3-Clause, I could probably consolidate the licenses, I guess...

Licensing is hard :/

**Comment by phyber** [comment]
Yeah, that's cool, thanks. I was about to suggest you just take it under the BSD-3-Clause :)

**Comment by phyber** [comment]
I'll definitely switch `jail_exporter` over to BSD-3-Clause later so any other code sharing is much simpler.

**Comment by phyber** [comment]
Finished reading over this, I think I understand it and it's looking good. :+1:

**Comment by fabianfreyer** [comment]
I've polished this up, and moved it to [fubarnetes/rctl](https://github.com/fubarnetes/rctl).

**Comment by phyber** [comment]
Nice. I'll move `jail_exporter` over to the `rctl` crate later today :)

**Comment by fabianfreyer** [comment]
no need; you can just call `jail.racct_statistics()` :) - I'll add some more boilerplate later and probably release 0.0.5 later :)

**Comment by fabianfreyer** [comment]
As tests seem to work locally, I'll just merge this.

**Pull request merged**